zcash_primitives/src/sapling/note_encryption.rs

```diff
@@ -254,16 +257,12 @@ impl<P: consensus::Parameters> Domain for SaplingDomain<P> {
     fn parse_note_plaintext_without_memo_ovk(
         &self,
         pk_d: &Self::DiversifiedTransmissionKey,
-        esk: &Self::EphemeralSecretKey,
-        ephemeral_key: &EphemeralKeyBytes,
+        _esk: &Self::EphemeralSecretKey,
+        _ephemeral_key: &EphemeralKeyBytes,
```

Comment:
It should be possible to delete these arguments.

Reply:
A secondary advantage of changing the signature of this method is that it becomes impossible to mistakenly link together code containing a caller of the old API that might (in principle) assume the stronger guarantee, and code containing an implementation of the new API with the weaker guarantee. This seems more robust, even though it doesn't apply to any actual code in our crates as far as we know. (Changes in contract *without* a corresponding change in type signature make me nervous in general.)

Reply:
Indeed, but as I said in the commit message, this needs to happen separately in sync with the `orchard` create, and I didn't want to block this PR on doing so.

Reply:
I don't think we need to be in any particular hurry to merge this; it can wait until we're ready to cut breaking releases of `zcash_note_encryption` and `orchard`. Note that it doesn't improve the "invalid" case of decryption which is the bottleneck in scanning, it only improves the "valid" case of decryption.

Reply:
Actually I'm fine to merge this and make the breaking change as a follow-up.